numpy/core/src/private/lowlevel_strided_loops.h

```diff
@@ -673,6 +691,60 @@ npy_bswap8_unaligned(char * x)
                                               NPY_ARRAY_F_CONTIGUOUS)) \
                         )
 
+#define PyArray_TRIVIAL_PAIR_ITERATION_STRIDE(size, arr) ( \
+                        size == 1 ? 0 : ((PyArray_NDIM(arr) == 1) ? \
+                                          PyArray_STRIDE(arr, 0) : \
+                                          PyArray_ITEMSIZE(arr)))
+
+static NPY_INLINE int
+PyArray_EQUIVALENTLY_ITERABLE_OVERLAP_OK(PyArrayObject *arr1, PyArrayObject *arr2,
```

Comment:
OTOH, I don't really want to put it in a .c file since the logic is tightly coupled to the macros here. Removing the inline makes it emit compiler warnings. I could write this as a macro, but that's messy.